static/app/views/releases/list/thresholdsList.tsx

```diff
@@ -0,0 +1,108 @@
+import {useMemo} from 'react';
+import styled from '@emotion/styled';
+
+import LoadingError from 'sentry/components/loadingError';
+import LoadingIndicator from 'sentry/components/loadingIndicator';
+import PanelTable from 'sentry/components/panels/panelTable';
+import {t} from 'sentry/locale';
+import {space} from 'sentry/styles/space';
+
+import {Threshold} from '../utils/types';
+import useFetchThresholdsListData from '../utils/useFetchThresholdsListData';
+
+import {ThresholdGroupRow} from './thresholdGroupRow';
+
+type Props = {
+  selectedEnvs: string[];
+  selectedProjectIds: number[];
+};
+
+function ThresholdsList({selectedEnvs, selectedProjectIds}: Props) {
+  const {
+    data: thresholds = [],
+    error,
+    isLoading,
+    isError,
+    refetch,
+  } = useFetchThresholdsListData({
+    selectedProjectIds,
+  });
+
+  // NOTE: currently no way to filter for 'None' environments
+  const filteredThresholds = selectedEnvs.length
+    ? thresholds.filter(
+        threshold => selectedEnvs.indexOf(threshold.environment.name) > -1
+      )
+    : thresholds;
+
+  const thresholdGroups: {[key: string]: {[key: string]: Threshold[]}} = useMemo(() => {
+    const byProj = {};
+    filteredThresholds.forEach(threshold => {
+      const projId = threshold.project.id;
+      if (!byProj[projId]) {
+        byProj[projId] = {};
+      }
+      const env = threshold.environment.name;
+      if (!byProj[projId][env]) {
+        byProj[projId][env] = [];
+      }
+      byProj[projId][env].push(threshold);
+    });
+    return byProj;
+  }, [filteredThresholds]);
```

Comment:
consider using lodash's groupBy since we already have it

Reply:
@scttcper 
QQ - this something you think should also avoid memoizing? Or would this be an appropriate use?
🤷 dunno if iteration is valuable to memoize or if really it should just be for operations like querying or something

Reply:
yeah this makes sense to memoize, could do groupBy for one of the layers or leave as is